app/src/main/java/org/p2p/wallet/moonpay/repository/sell/MoonpaySellRemoteRepository.kt

```diff
@@ -60,18 +64,36 @@ class MoonpaySellRemoteRepository(
     @Throws(MoonpaySellError::class)
     override suspend fun getUserSellTransactions(
         userAddress: Base58String
-    ): List<MoonpaySellTransaction> = doMoonpayRequest {
+    ): List<SellTransaction> = doMoonpayRequest {
+        val response = moonpayServerSideApi.getUserSellTransactions(userAddress.base58Value)
+        val depositWallets = getDepositWalletsForTransactions(response)
+
         mapper.fromNetwork(
-            response = moonpayServerSideApi.getUserSellTransactions(userAddress.base58Value),
-            transactionOwnerAddress = userAddress
+            response = response,
+            depositWallets = depositWallets,
+            selectedFiat = getSellFiatCurrency(),
+            transactionOwnerAddress = userAddress,
         )
     }
 
+    private suspend fun getDepositWalletsForTransactions(
+        transactions: List<MoonpaySellTransactionShortResponse>
+    ): List<MoonpaySellTransactionDepositWalletResponse> =
+        transactions.filter { it.status == SellTransactionStatus.WAITING_FOR_DEPOSIT }
+            .mapNotNull {
+                kotlin.runCatching {
+                    moonpayClientSideApi.getSellTransactionDepositWalletById(
+                        transactionId = it.transactionId,
+                        apiKey = BuildConfig.moonpayKey
+                    )
+                }.getOrNull() // skip if couldn't fetch
```

Comment:
imho, we need to change our code style rules about first operation. 
```
transactions
     .filter()
     .mapNotNull()
     .getOrNull()
```

is better

Reply:
Are you sure you need `runCatching` here?

Reply:
i need it, but im not sure how to do it properly, i'll come to dm